pkg/cmd/join/preflight/checks.go

```diff
@@ -0,0 +1,73 @@
+// Copyright Contributors to the Open Cluster Management project
+package preflight
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	clientcmdapiv1 "k8s.io/client-go/tools/clientcmd/api/v1"
+	"open-cluster-management.io/clusteradm/pkg/helpers"
+)
+
+type KlusterletApiserverCheck struct {
+	KlusterletApiserver string
+}
+
+func (c KlusterletApiserverCheck) Check() (warningList []error, errorList []error) {
```

Comment:
I see what's going on here. But you may want warnings as a []string to avoid confusion.

Reply:
done